pkg/orchestrator/interfaces.go

```diff
@@ -58,3 +59,23 @@ type EvaluationBroker interface {
 	// without having to wait for the dequeue visibility timeout.
 	Nack(evalID string, receiptHandle string) error
 }
+
+// Scheduler encapsulates the business logic of a scheduler. It processes
+// evaluations one at a time, generating task placements based on the provided
+// evaluation. The scheduler focuses on business logic, while other components handles
+// the underlying infrastructure and coordination between the orchestrator and compute nodes.
+type Scheduler interface {
+	// Process handles a new evaluation. It applies the necessary logic to determine
+	// task placements based on the provided evaluation.
+	Process(*model.Evaluation) error
+}
+
+// SchedulerProvider returns a scheduler instance that is capable of handling
+// jobs requiring scheduling.
+type SchedulerProvider interface {
+	// Scheduler returns a scheduler for the given job type
+	Scheduler(jobType string) Scheduler
```

Comment:
This should probably return an error?  Although this is tested as `unknown returns nil` it isn't immediately obvious from looking at this that `Scheduler` is an interface (and therefore nillable), and perhaps having an error makes it more explicit that it can fail. 

[aside] I really wish Go mandated that interfaces _had_ to be returned as `*Scheduler`, I bet it'd safe lots of nil panics.

Reply:
I was actually thinking about this and whether an error should be returned or not. My thinking was:

- If there can be different issues when returning a scheduler, then retuning an error with a descriptive message on what went wrong is helpful
- In this case, it is either a scheduler exists or does not exist. Thats why I thought of just returning a nullable result. However, this is based on `MappedSchedulerProvider` implementation 
- Another option could be `Scheduler(jobType string) (Scheduler, bool)`, where true means a scheduler was found

Overall, I need to fix this as it troubled me and you also raised a concern. I am just looking for the right _Go_ approach

Reply:
I went with returning an error